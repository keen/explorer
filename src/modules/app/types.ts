import {
  PickerWidgets,
  ChartSettings,
  WidgetSettings,
} from '@keen.io/widget-picker';

export type Confirmation = 'delete';

export type ViewMode = 'browser' | 'editor';

export enum SettingsModalSource {
  QUERY_SETTINGS,
  FIRST_QUERY_SAVE,
}

export type ReducerState = {
  confirmModal: {
    action: Confirmation;
    visible: boolean;
    meta?: Record<string, any>;
  };
  extractToEmailModal: {
    visible: boolean;
  };
  querySettingsModal: {
    visible: boolean;
    source: SettingsModalSource;
  };
  embedModal: {
    visible: boolean;
  };
  view: ViewMode;
  browserScreen: {
    width: number;
    height: number;
  };
  visualization: {
    type: PickerWidgets | null;
    chartSettings: ChartSettings;
    widgetSettings: WidgetSettings;
  };
  autorunQuery: boolean;
};
