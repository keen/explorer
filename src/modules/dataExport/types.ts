export type ReducerState = {
  exportCSVModalVisible: boolean;
};
