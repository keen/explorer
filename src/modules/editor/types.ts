import { MENU_ITEMS_ENUM } from '@keen.io/widget-customization';

export enum EditorSection {
  QUERY = 'query',
  SETTINGS = 'settings',
}

export type ReducerState = {
  activeEditorTab?: EditorSection;
  activeSettingsSection?: MENU_ITEMS_ENUM;
};
