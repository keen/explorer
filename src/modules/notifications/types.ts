export type ToastSettings = {
  autoDismiss?: boolean;
  showDismissButton?: boolean;
  translateMessage?: boolean;
  message: string;
  type: 'error' | 'success' | 'info';
};
