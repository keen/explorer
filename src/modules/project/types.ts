export type ReducerState = {
  tagsPool: string[];
};
