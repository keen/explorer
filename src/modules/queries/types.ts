import {
  PickerWidgets,
  ChartSettings,
  WidgetSettings,
} from '@keen.io/widget-picker';

import { APIError } from '../../types';

export type Visualization = {
  type: PickerWidgets;
  chartSettings: ChartSettings;
  widgetSettings: WidgetSettings;
};

export type SavedQueryListItem = {
  name: string;
  displayName: string;
  refreshRate: number;
  lastModifiedDate: string;
  cached: boolean;
  query: Record<string, any>;
  tags: string[];
  visualization: Visualization;
};

export type ReducerState = {
  results: any;
  settings: Record<string, any>;
  isQueryPerforming: boolean;
  isSavedQueriesLoaded: boolean;
  isSavingQuery: boolean;
  savedQueries: SavedQueryListItem[];
  cachedQueries: {
    limit: null | number;
    limitReached: boolean;
  };
  queriesExecution: {
    limitReached: boolean;
  };
  saveQueryError: APIError | null;
  error: APIError | null;
};
