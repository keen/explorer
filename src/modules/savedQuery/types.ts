export type SavedQuery = {
  name: string;
  displayName: string;
  cached: boolean;
  isCloned: boolean;
  tags: string[];
  refreshRate: number;
  exists: boolean;
};

export type ReducerState = SavedQuery;
