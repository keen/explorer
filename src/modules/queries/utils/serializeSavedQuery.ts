/* eslint-disable @typescript-eslint/camelcase */
import serializeVisualization from './serializeVisualization';

import { SavedQueryListItem } from '../types';
import { SavedQueryAPIResponse } from '../../../types';

const serializeSavedQuery = ({
  query_name,
  refresh_rate,
  user_last_modified_date,
  metadata,
  query,
}: SavedQueryAPIResponse): SavedQueryListItem => ({
  name: query_name,
  displayName: metadata ? metadata.display_name : query_name,
  cached: !!refresh_rate,
  refreshRate: refresh_rate,
  lastModifiedDate: user_last_modified_date,
  visualization: serializeVisualization(
    query,
    metadata && metadata.visualization
  ),
  tags: metadata && metadata.tags ? metadata.tags : [],
  query,
});

export default serializeSavedQuery;
