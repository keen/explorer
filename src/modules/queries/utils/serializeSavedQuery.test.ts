/* eslint-disable @typescript-eslint/camelcase */
import serializeSavedQuery from './serializeSavedQuery';

import { SavedQueryAPIResponse } from '../../../types';

test('serializes saved query', () => {
  const savedQuery: SavedQueryAPIResponse = {
    query_name: 'purchases',
    refresh_rate: 14000,
    user_last_modified_date: 'user_last_modified_date',
    created_date: 'created_date',
    last_modified_date: 'last_modified_date',
    query: { analysis_type: 'count' },
    metadata: {
      display_name: 'purchases',
      visualization: {
        type: 'bar',
        chart_settings: {
          layout: 'vertical',
          group_mode: 'grouped',
        },
        widget_settings: {},
      },
    },
  };

  expect(serializeSavedQuery(savedQuery)).toMatchSnapshot();
});
