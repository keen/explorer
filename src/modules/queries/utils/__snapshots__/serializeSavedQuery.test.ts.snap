// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serializes saved query 1`] = `
Object {
  "cached": true,
  "displayName": "purchases",
  "lastModifiedDate": "user_last_modified_date",
  "name": "purchases",
  "query": Object {
    "analysis_type": "count",
  },
  "refreshRate": 14000,
  "tags": Array [],
  "visualization": Object {
    "chartSettings": Object {
      "groupMode": "grouped",
      "layout": "vertical",
    },
    "type": "bar",
    "widgetSettings": Object {},
  },
}
`;
